document-portal/document-portal-fuse.c

```diff
@@ -3079,11 +3174,47 @@ xdp_fuse_mainloop (gpointer data)
       g_file_set_contents (status, s->str, -1, &error);
       g_assert_no_error (error);
     }
+}
 
-  fuse_session_remove_chan (main_ch);
-  fuse_session_destroy (session);
-  fuse_unmount (mount_path, main_ch);
-  main_ch = NULL;
+static gpointer
+xdp_fuse_thread (gpointer data)
+{
+  const char *path;
+  struct fuse_session *se;
+  XdpFuseThreadData *thread_data = data;
+  GError *error = NULL;
+
+  fuse_pthread = pthread_self ();
+
+  se = fuse_session_new (thread_data->args, &xdp_fuse_oper,
+                         sizeof (xdp_fuse_oper), NULL);
+  if (se == NULL)
+    {
+      g_set_error (&error, XDG_DESKTOP_PORTAL_ERROR,
+                   XDG_DESKTOP_PORTAL_ERROR_FAILED,
+                   "Can't create fuse session");
+      g_atomic_pointer_set (&thread_data->error, error);
+      return NULL;
+    }
+
+  path = xdp_fuse_get_mountpoint ();
+  if (fuse_session_mount (se, path) != 0)
+    {
+      fuse_session_destroy (se);
+      g_set_error (&error, XDG_DESKTOP_PORTAL_ERROR,
+                   XDG_DESKTOP_PORTAL_ERROR_FAILED,
+                   "Can't mount path %s", path);
+      g_atomic_pointer_set (&thread_data->error, error);
+      return NULL;
+    }
+
+  g_atomic_pointer_set (&session, se);
+  xdp_fuse_mainloop (session, &thread_data->loop_config);
```

Comment:
it shouldn't really matter much as that's what we've just created and given that the loop didn't begin yet when we're starting it shouldn't be racy, but indeed could be cleaner to use `se`.

Reply:
Ok done... also really I made sure that `session` is only set in one place to avoid confusion so using in the thread function one or another isn't really a big deal.